MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1697,36 +1697,62 @@ static void signalAndUntrack(const MVKSwapchainSignaler& signaler) {
 		default:
 			break;
 	}
+	
+	// If only stencil aspect is requested, possibly change to stencil-only format.
+	if (aspectMask == VK_IMAGE_ASPECT_STENCIL_BIT) {
+		if (_mtlPixFmt == MTLPixelFormatDepth32Float_Stencil8) {
+			_mtlPixFmt = MTLPixelFormatX32_Stencil8;
+		}
+#if MVK_MACOS
+		else if (_mtlPixFmt == MTLPixelFormatDepth24Unorm_Stencil8) {
+			_mtlPixFmt = MTLPixelFormatX24_Stencil8;
+		}
+#endif
+	}
```

Comment:
The original code had a check that the view usage included `SAMPLED` or `INPUT_ATTACHMENT` before switching the format. You should probably check that here, too.

Reply:
Will do! When I started the review I was not sure why the check was there, but after some research I see it's required. I will add a comment in code for the future.